File: Problem 2/problem2.js
const numbers = [1, 2, 3, 4, 5];

function solve(arr) {
  const newArr = [];
  for (i = 0; i < arr.length; i++) {
    let aux = arr[i];
    delete arr[i];
    newArr[i] = arr.reduce((a, b) => a * b);
    arr[i] = aux;
  }
  return console.log(newArr);
}

solve(numbers);

File: Problem 2/problem2.py
import unittest


def solve(arr):
    new_arr = [reduce(lambda x, y: x * y, arr[:i] + arr[i+1:])
               for i in range(len(arr))]
    return new_arr


class Test(unittest.TestCase):
    def test(self):
        attempt = solve([1, 2, 3, 4, 5])
        expected = [120, 60, 40, 30, 24]
        self.assertEquals(attempt, expected)


if __name__ == '__main__':
    unittest.main()
